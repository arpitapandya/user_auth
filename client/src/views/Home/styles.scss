
.home_banner {
  margin-top: 40px;
  padding: 20px;
  text-align: center;
  max-width: 700px;
  margin: auto;
  .details{
    font-weight: 600;
    font-size: 18px;
  }
  .main_subtitle {
    text-align: center;
    padding: 24px 16px;
    font-weight: 600;
    font-size: 28px;
  }
  .input_banner {
    margin-bottom: 16px;
    label {
      font-size: 14px;
      font-weight: 600;
      margin: 16px 0 4px 0;
    }
    input {
      box-shadow: none;
      border-radius: 2px;
      border: 1px solid #dbdbdb;
      background-color: #fafafa !important;
      padding: 0 10px;
      width: 100%;
      outline: 0;
      height: 32px;
      box-sizing: border-box;
      margin-top: 4px;
    }
    .error-text {
      padding: 0;
      margin: 0;
      font-size: 14px;
      padding-left: 2px;
    }
  }
  .btn_banner {
    text-align: center;
    button {
      padding: 10px;
      font-size: 18px;
      font-weight: 600;
      border-radius: 2px;
      cursor: pointer;
      width: 100%;
      background-color: #3897f0;
      outline: none;
      color: #fff;
      border: none;
      outline: none;
      font-size: 16px;
      margin-top: 10px;
    }
  }
}
