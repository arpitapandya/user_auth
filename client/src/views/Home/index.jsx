/* eslint-disable no-alert */
import React, {useState, useEffect} from 'react';
import { useDispatch, useSelector } from 'react-redux'

import {changePasswordAdmin} from 'api/user'
import { changePasswordSuccess, changePasswordFailure, startChangePassword } from 'actions/user';


import './styles.scss';

function Home(props) {
  const dispatch = useDispatch();
  const auth = useSelector(state => state.auth)
  const user = useSelector(state => state.auth.user);

  const [password, setPassword] = useState('');
  const [isSubmit, setSubmit] = useState(false);

  useEffect(() => {
    if (!auth.isChangePasswordApiHit && isSubmit) {
      if (auth.isChangePasswordError) {
        alert(auth.isChangePasswordError)
      } else {
        alert('Password Changed Successfully');
        setPassword('');
      }
      setSubmit(false)
    }
  }, [auth.isChangePassword, auth.isChangePasswordApiHit, auth.isChangePasswordError])

  const displayUser = props?.location?.state ? props?.location?.state : user;

  const updatePassword = () => {
    const data = {
      newPassword: password,
      username: displayUser.username,
    }
    setSubmit(true);
    dispatch(startChangePassword());
    dispatch(changePasswordAdmin(data, changePasswordSuccess, changePasswordFailure));
  }

  return (
    <div className='home_banner'>
      <div className='details'>USER NAME: {displayUser?.username}</div>
      <div className='details'>USER AGE: {displayUser?.age}</div>

      {user.role === 'admin' && (
        <>
          <div className='input_banner'>
            <label htmlFor='password'>Password (8 or more characters)</label>
            <input
              name='password'
              type='password'
              value={password}
              autoComplete='off'
              onChange={(e) => setPassword(e.target.value)}
            />
          </div>
          <div className='btn_banner'>
            <button disabled={!password} type='button' onClick={updatePassword}>
              Change Password
            </button>
          </div>
        </>
      )}

    </div>
  );
}

export default Home;
