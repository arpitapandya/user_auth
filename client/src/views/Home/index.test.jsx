import React from 'react';
import renderer from 'react-test-renderer';
import Home from './index';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

describe('My Connected React-Redux Component', () => {
  let store;
  let component;
 
  beforeEach(() => {
    store = mockStore({
      auth: {
        user: {},
        isLoggedIn: false,
        error: null,
        isChangePassword: false,
        isChangePasswordError: null,
        isChangePasswordApiHit: false,
      },
      users: [],
    });
  });
 
  it('renders correctly', () => {
    const tree = renderer
      .create(<Provider store={store}>
        <Home />
      </Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});


