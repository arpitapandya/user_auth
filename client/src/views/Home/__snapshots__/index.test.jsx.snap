// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component renders correctly 1`] = `
<div
  className="home_banner"
>
  <div
    className="details"
  >
    USER NAME: 
  </div>
  <div
    className="details"
  >
    USER AGE: 
  </div>
</div>
`;
